hopsanc/src/hopsanc.cpp

```diff
@@ -0,0 +1,282 @@
+#include "hopsanc.h"
+#include <iostream>
+#include <string.h>
+
+#include "HopsanCore.h"
+#include "HopsanEssentials.h"
+#include "ComponentSystem.h"
+
+static hopsan::ComponentSystem *spCoreComponentSystem = nullptr;
+static hopsan::HopsanEssentials gHopsanCore;
+
+static double startTime, stopTime;
+
+//! @brief Prints all waiting messages
+//! @param[in] printDebug Should debug messages also be printed
+void printWaitingMessages(hopsan::HopsanEssentials& hopsanCore, bool printDebug, bool silent)
+{
+    if(silent) return;
+
+    hopsan::HString msg, type, tag;
+    while (hopsanCore.checkMessage() > 0) {
+        hopsanCore.getMessage(msg,type,tag);
+        if (type == "debug") {
+            if (printDebug) {
+                std::cout << msg.c_str() << "\n";
+            }
+        }
+        else {
+            std::cout << msg.c_str() << "\n";
+        }
+    }
+}
+
+
+int loadModel(const char* path) {
+    if(spCoreComponentSystem) {
+        delete spCoreComponentSystem;
+    }
+    spCoreComponentSystem = gHopsanCore.loadHMFModelFile(path, startTime, stopTime);
+    if(!spCoreComponentSystem) {
+        std::cout << "Failed to instantiate model!\n";
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    }
+    const hopsan::HString modelName = spCoreComponentSystem->getName();
+    spCoreComponentSystem->addSearchPath(modelName+"-resources");
+    std::cout << "Loaded model: " << modelName.c_str() << "\n";
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+
+//! @brief Returns specified data vector as C array
+//! @param [in] variable Variable name ("component.port.variable")
+//! @param [out] size Size of returned data array
```

Comment:
update documentation string

Reply:
I thought about using a size argument, but it felt weird to have an argument that is only allowed to have a specified value. I think it is better to clearly define the requirements on the buffer size in the documentation.